core/node/libp2p/transport.go

```diff
@@ -31,15 +32,24 @@ func Transports(tptConfig config.Transports) interface{} {
 
 		if tptConfig.Network.QUIC.WithDefault(!privateNetworkEnabled) {
 			if privateNetworkEnabled {
-				// QUIC was force enabled while the private network was turned on.
-				// Fail and tell the user.
 				return opts, fmt.Errorf(
 					"The QUIC transport does not support private networks. " +
 						"Please disable Swarm.Transports.Network.QUIC.",
 				)
 			}
 			// TODO(9290): Make WithMetrics configurable
-			opts.Opts = append(opts.Opts, libp2p.Transport(libp2pquic.NewTransport, libp2pquic.WithMetrics()))
+			opts.Opts = append(opts.Opts, libp2p.Transport(quic.NewTransport, quic.WithMetrics()))
+		}
+
+		// TODO(9292): Remove the false && to allows it enabled by default
+		if tptConfig.Network.WebTransport.WithDefault(false && !privateNetworkEnabled) {
```

Comment:
nit:
```suggestion
		if tptConfig.Network.WebTransport.WithDefault(false) && !privateNetworkEnabled {
```

Reply:
No no, look at how QUIC is implemented above, this is correct.

(I know the `false &&` is cursed, this is just to help us remind what we are supposed to do `!privateNetworkEnabled` but to also temporarily disable this check)

Reply:
this will be correct when WebTransport will be true by default. Right now is not the case. We can always change the code to fit the needed use case.